diff --git a/drivers/media/pci/ddbridge/ddbridge.c b/drivers/media/pci/ddbridge/ddbridge.c
index fab421f..031288a 100644
--- a/drivers/media/pci/ddbridge/ddbridge.c
+++ b/drivers/media/pci/ddbridge/ddbridge.c
@@ -109,7 +109,7 @@ static void ddb_irq_msi(struct ddb *dev, int nr)
 	int stat;
 
 	if (msi && pci_msi_enabled()) {
-		stat = pci_alloc_irq_vectors(dev->pdev, 1, nr, PCI_IRQ_MSI);
+		stat = pci_enable_msi_range(dev->pdev, 1, nr);
 		if (stat >= 1) {
 			dev->msi = stat;
 			pr_info("using %d MSI interrupt(s)\n",
